server/routerlicious/packages/routerlicious-base/src/nexus/runnerFactory.ts

```diff
@@ -0,0 +1,533 @@
+		winston.info(`tenant orderer: ${JSON.stringify(tenant.orderer)}`, { messageMetaData });
+		Lumberjack.info(
+			`tenant orderer: ${JSON.stringify(tenant.orderer)}`,
+			getLumberBaseProperties(documentId, tenantId),
+		);
+
+		if (tenant.orderer.url !== this.ordererUrl && !this.globalDbEnabled) {
+			Lumberjack.error(`Invalid ordering service endpoint`, { messageMetaData });
+			throw new Error("Invalid ordering service endpoint");
+		}
+
+		switch (tenant.orderer.type) {
+			case "kafka":
+				return this.kafkaFactory.create(tenantId, documentId);
+			default:
+				return this.localOrderManager.get(tenantId, documentId);
+		}
+	}
+}
+
+export class NexusResources implements core.IResources {
+	public webServerFactory: core.IWebServerFactory;
+
+	constructor(
+		public config: Provider,
+		public redisConfig: any,
+		public clientManager: core.IClientManager,
+		public webSocketLibrary: string,
+		public orderManager: core.IOrdererManager,
+		public tenantManager: core.ITenantManager,
+		public socketConnectTenantThrottler: core.IThrottler,
+		public socketConnectClusterThrottler: core.IThrottler,
+		public socketSubmitOpThrottler: core.IThrottler,
+		public socketSubmitSignalThrottler: core.IThrottler,
+		public singleUseTokenCache: core.ICache,
+		public storage: core.IDocumentStorage,
+		public mongoManager: core.MongoManager,
+		public port: any,
+		public documentsCollectionName: string,
+		public metricClientConfig: any,
+		public throttleAndUsageStorageManager?: core.IThrottleAndUsageStorageManager,
+		public verifyMaxMessageSize?: boolean,
+		public redisCache?: core.ICache,
+		public socketTracker?: core.IWebSocketTracker,
+		public tokenRevocationManager?: core.ITokenRevocationManager,
+		public revokedTokenChecker?: core.IRevokedTokenChecker,
+	) {
+		const socketIoAdapterConfig = config.get("nexus:socketIoAdapter");
+		const httpServerConfig: services.IHttpServerConfig = config.get("system:httpServer");
+		const socketIoConfig = config.get("nexus:socketIo");
+		this.webServerFactory = new services.SocketIoWebServerFactory(
+			this.redisConfig,
+			socketIoAdapterConfig,
+			httpServerConfig,
+			socketIoConfig,
+		);
+	}
+
+	public async dispose(): Promise<void> {
+		const mongoClosedP = this.mongoManager.close();
+		const tokenRevocationManagerP = this.tokenRevocationManager
+			? this.tokenRevocationManager.close()
+			: Promise.resolve();
+		await Promise.all([mongoClosedP, tokenRevocationManagerP]);
+	}
+}
+
+export class NexusResourcesFactory implements core.IResourcesFactory<NexusResources> {
+	public async create(
+		config: Provider,
+		customizations?: IAlfredResourcesCustomizations,
+	): Promise<NexusResources> {
+		const metricClientConfig = config.get("metric");
+		// Producer used to publish messages
+		const kafkaEndpoint = config.get("kafka:lib:endpoint");
+		const kafkaLibrary = config.get("kafka:lib:name");
+		const kafkaClientId = config.get("alfred:kafkaClientId");
```

Comment:
it should be nexus:kafkaClientId

Reply:
Will leave as is, deli also depends on alfred:kafkaClientId, they need to all be in sync. For next PR we can move all to Nexus (taking care of n/n-1)

Reply:
Upon further investigation. We can safely change to nexus, it is only used for telemetry.